### PR TITLE
chore: add contract test util typedoc entry point

### LINKED ISSUE #2919: Typedoc produces unnecessary docs when used with multiple entry points
Following https://github.com/FuelLabs/fuels-ts/pull/2082 we are now able to support multiple entry points in a _typedoc_ configuration, but when one passes multiple entry points it results in [unnecessary files](https://github.com/FuelLabs/fuels-ts/pull/1356#discussion_r1593712213) which contain the exported types along with the relevant typedocs.

E.g. adding `test-utils.ts` to the [contract typedoc](https://github.com/FuelLabs/fuels-ts/blob/master/packages/contract/typedoc.json#L4)

```ts
{
  "$schema": "https://typedoc.org/schema.json",
  "extends": ["../../typedoc.base.json"],
  "entryPoints": ["src/index.ts", "src/test-utils.ts"]
}
```

Results in  the screenshot below:

Screenshot 2024-08-12 at 6 21 16 PM

### PR DESCRIPTION
**Pull request by danielbate**
- Closes #2919 
- Unblocks #1898 

# Summary

Adds the test util type doc entry point for the contract package and removes redundant sidebar links.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: apps/docs/scripts/typedoc-postbuild.ts [hunk 1]
```diff
@@ -65,7 +65,12 @@ const exportLinksJson = () => {
     .forEach((directory) => {
       links.items.push({ text: directory, link: `/api/${directory}/`, collapsed: true, items: [] });
       readdirSync(join(apiDocsDir, directory))
-        .filter((file) => file !== 'index.md')
+        .filter(
```
Comment (maschad) [inline apps/docs/scripts/typedoc-postbuild.ts:68]: Pardon my ignorance as I'm not super familiar with this postbuild script but I do get the impression we probably should do a wider refactor on it. 

I think we can make this a bit more flexible though should someone want to add other file names within multiple entry points

```suggestion
.filter((file) => {
          // Exclude index files and files related to secondary entry points
          const isIndexFile = file.endsWith('index.md');
          const isSecondaryEntryPoint = secondaryEntryPoints.some((entryPoint) =>
            file.includes(entryPoint.replace('-', '-').replace('.md', ''))
          );
          return !isIndexFile && !isSecondaryEntryPoint;
        })
```

Comment (danielbate) [reply apps/docs/scripts/typedoc-postbuild.ts]: I do agree, although I think it's been simplified heavily with the new typedoc version as in #2650.

Comment (petertonysmith94) [reply apps/docs/scripts/typedoc-postbuild.ts]: I'm personally wondering whether we want to move the API docs out of the docs hub and hosting this elsewhere. As it adds a lot of noise to the [docs](https://docs.fuel.network/docs/fuels-ts/api/).

**Comment by petertonysmith94** [comment]
Does this close https://github.com/FuelLabs/fuels-ts/issues/1898 as well?

**Comment by danielbate** [comment]
@petertonysmith94 it wont close it but it will unblock it.

File: apps/docs/scripts/typedoc-postbuild.ts (hunk 1 above)
Comment (danielbate) [reply apps/docs/scripts/typedoc-postbuild.ts]: I agree tbh, the default typedoc styling is really [clean](https://typedoc.org/example/) and it bugs me some time that the docs hub search includes all the api doc links.

File: apps/docs/scripts/typedoc-postbuild.ts [hunk 2]
```diff
@@ -65,7 +65,14 @@
     .forEach((directory) => {
       links.items.push({ text: directory, link: `/api/${directory}/`, collapsed: true, items: [] });
       readdirSync(join(apiDocsDir, directory))
-        .filter((file) => file !== 'index.md')
+        .filter((file) => {
+          // Exclude index files and files related to secondary entry points
+          const isIndexFile = file.endsWith('index.md');
+          const isSecondaryEntryPoint = secondaryEntryPoints.some((entryPoint) =>
+            file.includes(entryPoint.replace('-', '-').replace('.md', ''))
```
[the remark this answers is not in the document]
Comment (petertonysmith94) [reply apps/docs/scripts/typedoc-postbuild.ts:72]: @danielbate this replacement is redundant :)

**Review by petertonysmith94 — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by nedsalk — APPROVED** [review]

**Pull request merged**